source/StellarObject.h

```diff
@@ -14,6 +14,7 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 #define STELLAR_OBJECT_H_
 
 #include "Body.h"
+#include "PlayerInfo.h"
```

Comment:
This should be a forward declaration and not an include, since the class is simply "mentioned" in its interface, rather than kept in an object. The implementation, then, would need the `#include`, since it is accessing properties of the class.

(`#include`ing it in the header means whenever the PlayerInfo interface changes, any compilation unit that includes the StellarObject header will also need to be recompiled, while having it as a forward declaration would not, since the StellarObject header is basically telling units that include it, "I consume this PlayerInfo thing, but you don't need to worry about how I do that")